includes/class-scheduler.php

```diff
@@ -321,6 +329,40 @@ public static function purge_outbox() {
 		}
 	}
 
+	/**
+	 * Purge inbox items based on a schedule.
+	 */
+	public static function purge_inbox() {
+		$total_posts = (int) wp_count_posts( Inbox::POST_TYPE )->publish;
+		if ( $total_posts <= 200 ) {
+			return;
+		}
+
+		$days     = (int) get_option( 'activitypub_inbox_purge_days', 180 );
+		$timezone = new \DateTimeZone( 'UTC' );
+		$date     = new \DateTime( 'now', $timezone );
+
+		$date->sub( \DateInterval::createFromDateString( "$days days" ) );
+
+		$post_ids = \get_posts(
+			array(
+				'post_type'   => Inbox::POST_TYPE,
+				'post_status' => 'any',
+				'fields'      => 'ids',
+				'numberposts' => -1,
+				'date_query'  => array(
+					array(
+						'before' => $date->format( 'Y-m-d' ),
+					),
+				),
+			)
```

Comment:
```suggestion
		$days     = (int) \get_option( 'activitypub_inbox_purge_days', 180 );
		$post_ids = \get_posts(
			array(
				'post_type'   => Inbox::POST_TYPE,
				'post_status' => 'any',
				'fields'      => 'ids',
				'numberposts' => -1,
				'date_query'  => array(
					array(
						'before' => gmdate( 'Y-m-d', time() - ( $days * DAY_IN_SECONDS ) ),
					),
				),
			)
```
Would this work? Might be a bit simpler.

Reply:
let's see what the tests say!

Reply:
A different level of cowboy coding 😄